module/data/item/fields/enchantment-field.mjs

```diff
@@ -123,6 +143,59 @@ export class EnchantmentData extends foundry.abstract.DataModel {
 
     return errors.length ? errors : true;
   }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch the tracked enchanted items.
+   * @returns {Promise<Item5e[]>}
+   */
+  async getItems() {
+    return (await Promise.all(
+      Array.from(this.constructor.#appliedEnchantments.get(this.item.uuid) ?? [])
+        .map(uuid => fromUuid(uuid).then(ae => ae?.parent))
+    )).filter(i => i);
+  }
+
+  /* -------------------------------------------- */
+  /*  Static Registry                             */
+  /* -------------------------------------------- */
+
+  /**
+   * Registration of enchanted items mapped to a specific enchantment source. The map is keyed by the UUID of
+   * enchantment sources while the set contains UUID of items that source has enchanted.
+   * @type {Map<string, Set<string>>}
+   */
+  static #appliedEnchantments = new Map();
+
+  /* -------------------------------------------- */
+
+  /**
+   * Add a new enchantment effect to the list of tracked enchantments. Will not track enchanted items in compendiums.
+   * @param {string} source     UUID of the active effect origin for the enchantment.
+   * @param {string} enchanted  UUID of the enchantment to track.
+   */
+  static trackEnchantment(source, enchanted) {
+    if ( enchanted.startsWith("Compendium.") ) return;
+    if ( !this.#appliedEnchantments.has(source) ) {
+      this.#appliedEnchantments.set(source, new Set());
+    }
+    const applied = this.#appliedEnchantments.get(source);
+    applied.add(enchanted);
+  }
```

Comment:
This is a neat way of doing it that would also serve concentration effects and other buffs in general well.